src/vs/workbench/contrib/chat/browser/chatAttachmentModel/chatInstructionAttachmentsModel.ts

```diff
@@ -42,6 +76,48 @@ export class ChatInstructionAttachmentsModel extends Disposable {
 		return result;
 	}
 
+	/**
+	 * Get the list of all prompt instruction attachment variables, including all
+	 * nested child references of each attachment explicitly attached by user.
+	 */
+	public get chatAttachments(): readonly IChatRequestVariableEntry[] {
+		const result = [];
+		const attachments = [...this.attachments.values()];
+
+		for (const attachment of attachments) {
+			const { reference } = attachment;
+
+			// the usual URIs list of prompt instructions is `bottom-up`, therefore
+			// we do the same herfe - first add all child references of the model
+			result.push(
+				...reference.allValidReferencesUris.map((uri) => {
+					return toChatVariable('vscode.prompt.instructions', uri);
+				}),
+			);
+
+			// then add the root reference of the model itself
+			result.push(
+				toChatVariable('vscode.prompt.instructions.root', reference.uri),
```

Comment:
IDs are expected to be unique, for the children maybe you want a prefix plus the URI, similar to what we do for `#file` variables? Same goes for the root files if there can be multiple root prompt files in a single chat request

Reply:
Updated 👍